### ISSUE TITLE
[openssl-unix] Fails for android triplets, building on Windows when NDK is in a folder with '('

### ISSUE THREAD

**Issue by mschofie** [microsoft/vcpkg#13745]
**Host Environment**
- OS: Windows, AndroidNDK under "C:\Program Files (x86)"

**To Reproduce**
With the ANDROID_NDK_HOME environment variable set to a folder under "C:/Program Files (x86)" the openssl-unix port fails for Android. It _looks like_ [`openssl-unix/CMakeLists.txt`](/microsoft/vcpkg/blob/4cbbcbddfd42d94d6d8ee6c8b6ad00ef6a0927d3/ports/openssl-unix/CMakeLists.txt#L138) sets `CC`, `LD`, etc.. without quoting them, and quoting them _might_ fix this. I haven't tried; I worked around by bringing my own NDK into a compliant folder.

**Failure logs** 
The relevant piece of 'install-arm-android-rel-out' is:

```
[2/3] cmd.exe /C "cd /D D:\a\1\s\vcpkg\buildtrees\openssl-unix\arm-android-rel\1.1.1g-bb2a9a5801.clean && set PATH=D:/a/1/s/vcpkg/downloads/tools/msys2/msys64/usr/bin;%PATH% && "C:\Program Files\CMake\bin\cmake.exe" -E touch D:/a/1/s/vcpkg/buildtrees/openssl-unix/arm-android-rel/1.1.1g-bb2a9a5801.clean/krb5.h && D:\a\1\s\vcpkg\downloads\tools\msys2\msys64\usr\bin\make.exe build_libs"
FAILED: CMakeFiles/build_libs 1.1.1g-bb2a9a5801.clean/libssl.a 1.1.1g-bb2a9a5801.clean/libcrypto.a 
cmd.exe /C "cd /D D:\a\1\s\vcpkg\buildtrees\openssl-unix\arm-android-rel\1.1.1g-bb2a9a5801.clean && set PATH=D:/a/1/s/vcpkg/downloads/tools/msys2/msys64/usr/bin;%PATH% && "C:\Program Files\CMake\bin\cmake.exe" -E touch D:/a/1/s/vcpkg/buildtrees/openssl-unix/arm-android-rel/1.1.1g-bb2a9a5801.clean/krb5.h && D:\a\1\s\vcpkg\downloads\tools\msys2\msys64\usr\bin\make.exe build_libs"
/usr/bin/perl "-I." -Mconfigdata "util/dofile.pl" \
    "-oMakefile" include/crypto/bn_conf.h.in > include/crypto/bn_conf.h
/usr/bin/perl "-I." -Mconfigdata "util/dofile.pl" \
    "-oMakefile" include/crypto/dso_conf.h.in > include/crypto/dso_conf.h
/usr/bin/perl "-I." -Mconfigdata "util/dofile.pl" \
    "-oMakefile" include/openssl/opensslconf.h.in > include/openssl/opensslconf.h
/usr/bin/make depend && /usr/bin/make _build_libs
make[1]: Entering directory '/d/a/1/s/vcpkg/buildtrees/openssl-unix/arm-android-rel/1.1.1g-bb2a9a5801.clean'
make[1]: Leaving directory '/d/a/1/s/vcpkg/buildtrees/openssl-unix/arm-android-rel/1.1.1g-bb2a9a5801.clean'
make[1]: Entering directory '/d/a/1/s/vcpkg/buildtrees/openssl-unix/arm-android-rel/1.1.1g-bb2a9a5801.clean'
C:/Program Files (x86)/Android/android-sdk/ndk-bundle/toolchains/llvm/prebuilt/windows-x86_64/bin/clang.exe  -I. -Iinclude -fPIC -pthread -Wall -O3 -Wno-error=unused-command-line-argument -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -march=armv7-a -mthumb -Wformat -Werror=format-security  --target=armv7-none-linux-androideabi21 --gcc-toolchain=C:/Program Files (x86)/Android/android-sdk/ndk-bundle/toolchains/llvm/prebuilt/windows-x86_64 --sysroot=C:/Program Files (x86)/Android/android-sdk/ndk-bundle/toolchains/llvm/prebuilt/windows-x86_64/sysroot -DOPENSSL_USE_NODELETE -DOPENSSL_PIC -DOPENSSLDIR="\"/etc/ssl\"" -DENGINESDIR="\"D:/a/1/s/vcpkg/packages/openssl-unix_arm-android/lib/engines-1.1\"" -DNDEBUG   -c -o apps/app_rand.o apps/app_rand.c
/bin/sh: -c: line 0: syntax error near unexpected token `('
/bin/sh: -c: line 0: `C:/Program Files (x86)/Android/android-sdk/ndk-bundle/toolchains/llvm/prebuilt/windows-x86_64/bin/clang.exe  -I. -Iinclude -fPIC -pthread -Wall -O3 -Wno-error=unused-command-line-argument -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -march=armv7-a -mthumb -Wformat -Werror=format-security  --target=armv7-none-linux-androideabi21 --gcc-toolchain=C:/Program Files (x86)/Android/android-sdk/ndk-bundle/toolchains/llvm/prebuilt/windows-x86_64 --sysroot=C:/Program Files (x86)/Android/android-sdk/ndk-bundle/toolchains/llvm/prebuilt/windows-x86_64/sysroot -DOPENSSL_USE_NODELETE -DOPENSSL_PIC -DOPENSSLDIR="\"/etc/ssl\"" -DENGINESDIR="\"D:/a/1/s/vcpkg/packages/openssl-unix_arm-android/lib/engines-1.1\"" -DNDEBUG   -c -o apps/app_rand.o apps/app_rand.c'
make[1]: *** [Makefile:662: apps/app_rand.o] Error 1
make[1]: Leaving directory '/d/a/1/s/vcpkg/buildtrees/openssl-unix/arm-android-rel/1.1.1g-bb2a9a5801.clean'
make: *** [Makefile:175: build_libs] Error 2
ninja: build stopped: subcommand failed.
```

**Comment by PhoebeHui** [comment]
@mschofie, would you like to summit a PR for this issue?

**Comment by mschofie** [comment]
I'll take a look, but I've worked-around it myself, so it's less of a priority for me. I was posting mainly to help other people out. I'll see if I can get to it this week. Thanks!

**Comment by FrankXie05** [comment]
@mschofie Are you still following this issue? You said you have solved this problem, can you be more specific?

**Comment by mschofie** [comment]
Not really following the issue, as I found a workaround of not using an Android NDK installed into a folder with a `(`. The issue might be less important these days. IIRC, Visual Studio can install an Android NDK, it's installed in the Visual Studio location and prior to VS2022 that was in the, say, `C:\Program Files (x86)\Microsoft Visual Studio\` folder on an x64 machine. When building the openssl-unix port, the Toolchain we used discovered the VS-installed Android NDK and specified paths as necessary. As the port was running configure, the `(` in the path wasn't sufficiently escaped/quoted so that cryptic errors were logged and the process failed.
